tensornetwork/tests/network_test.py

```diff
@@ -320,6 +320,66 @@ def test_flatten_edges_different_nodes_value_error(backend):
     tn.flatten_edges([e1, e2])
 
 
+def test_split_trace_edge(backend):
+  a = tn.Node(np.zeros((2, 6, 4, 6, 5, 5)), backend=backend)
+  c = tn.Node(np.zeros((2, 4)), backend=backend)
+  e1 = tn.connect(a[1], a[3])
+  e2 = tn.connect(a[4], a[5])
+  external_1 = tn.connect(a[0], c[0])
+  external_2 = tn.connect(c[1], a[2])
+  shape = (2, 1, 3)
+  new_edge_names = ["New Edge 2", "New Edge 1", "New Edge 3"]
+  new_edges = tn.split_edge(e1, shape, new_edge_names)  
+  assert a.shape == (2, 4, 5, 5) + shape + shape
+  assert a.edges == [external_1, external_2, e2, e2, *new_edges, *new_edges]
+  for i, new_edge in enumerate(new_edges):
+    assert new_edge.name == new_edge_names[i]
+  tn.check_correct({a, c})
+
+
+def test_split_edges_standard(backend):
```

Comment:
Can you add a test where you split a standard edge and then run `contract_between` and make sure the resulting tensor values are still correct?

Reply:
Thank you for this. This actually exposed a bug (I forgot to update the `axis_names` in `split_edge` leading to problems when calling `_remove_edges` in `contract_between`).

Reply:
And another bug: `reorder_axes` doesn't permute the axis names like `reorder_edges` does.

Reply:
Good catch!